Add ktlint

jsarabia opened the pull request:
Sets up ktlint, also updates gradle and kotlin, as needed

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/wycliffeassociates/otter-common/119)
<!-- Reviewable:end -->

Commit:
Add ktlint

Sets up ktlint, also updates gradle and kotlin, as needed

aunger reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-common/119)* status: 0 of 2 files reviewed, 1 unresolved discussion (waiting on @aunger and @sarabiaj)

___
*[build.gradle, line 26 at r1](https://reviewable.io/reviews/wycliffeassociates/otter-common/119#-LmpbvN27z0sw7vAUOYR:-LmpbvN34fXU9Wkfb7CZ:bg7xayi) ([raw file](https://github.com/wycliffeassociates/otter-common/blob/a8ae68b9150f5df55147d9f7df2f223ba4161167/build.gradle#L26)):*
> ```Gradle
> 
> apply plugin: 'kotlin'
> apply plugin: "org.jlleitschuh.gradle.ktlint"
> ```

Nit: quote consistency


<!-- Sent from Reviewable.io -->

aunger reviewed (approved):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/wycliffeassociates/otter-common/119)* status: all files reviewed, 1 unresolved discussion (waiting on @sarabiaj)



<!-- Sent from Reviewable.io -->

jsarabia committed:
Fix: Reassignment of read-only property via backing field (#118)

Changed val to var. Updating kotlin seemed to change this from a warning (which was suppressed) to an error.

Removed the suppress annotation as the field is now a var.

Commit:
plugin quote consistency

changes double quote to single for consistency
